### PR TITLE
adding dns name label support for azure cli container module

### PR DESCRIPTION
**Pull request by xizhamsft**
### With this change, user can now specify a dns name label for the container they create in ACI using Azure CLI. For example, az container create --dns-name-label=dnsnametest ...  The name conflict and syntax checks will be handled on the server side. 

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [X] Each command and parameter has a meaningful description.
- [X] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5487
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_help.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
           text: az container create -g MyResourceGroup --name mywinapp --image winappimage:latest --os-type Windows --cpu 2 --memory 3.5
         - name: Create a container in a container group with public IP address and ports.
           text: az container create -g MyResourceGroup --name myalpine --image alpine:latest --ip-address public --ports 80 443
+        - name: Create a container in a container group with public IP address, ports and DNS name label.
+          text: az container create -g MyResourceGroup --name myalpine --image alpine:latest --ip-address public --ports 80 443 --dns_name_label contoso
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/_help.py]: Should be `--dns-name-label` not `--dns_name_label`

**Comment by xizhamsft** [comment]
@derekbekoe , please take a look and let me know if there are any issues.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_help.py (hunk 1 above)
Comment (xizhamsft) [reply src/command_modules/azure-cli-container/azure/cli/command_modules/container/_help.py]: @derekbekoe , good catch. Fixed now.

**Comment by derekbekoe** [comment]
LGTM.
Resolve conflicts from https://github.com/Azure/azure-cli/commit/69e454a69f2547c90aa6c837ba01ba4db2830b3c then can merge.

**Pull request merged**